src/Components/TestExplorer.fs

```diff
@@ -446,16 +452,50 @@ module DotnetCli =
             else
                 promise { return (None, "", "") }
 
+
     let restore
         (projectPath: string)
         : JS.Promise<Node.ChildProcess.ExecError option * StandardOutput * StandardError> =
         Process.exec "dotnet" (ResizeArray([| "restore"; projectPath |]))
 
+    let private debugProcessIdRegex = RegularExpressions.Regex(@"Process Id: (.*),")
+
+    let private tryGetDebugProcessId consoleOutput =
+        let m = debugProcessIdRegex.Match(consoleOutput)
+
+        if m.Success then
+            let processId = m.Groups.[1].Value
+            Some processId
+        else
+            None
+
+    let private launchDebugger processId =
+        let launchRequest: DebugConfiguration =
+            {| name = ".NET Core Attach"
+               ``type`` = "coreclr"
+               request = "attach"
+               processId = processId |}
+            |> box
+            |> unbox
+
+        let folder = workspace.workspaceFolders.Value.[0]
+
+        promise {
+            let! _ =
+                Vscode.debug.startDebugging (Some folder, U2.Case2 launchRequest)
+                |> Promise.ofThenable
+
+            do! Promise.sleep 2000
```

Comment:
Have to wait or it'll continue before the debugger reaches the stop on entry point.
That'll leave the debugger in a confusing state where it shows it's attached but no breakpoints are hit and the breakpoints show as disabled

Reply:
Definitely leave a code comment for this one then.

Reply:
Done. 
Also commented on the processId weirdness